src/uct/ib/base/ib_md.c

```diff
@@ -1431,17 +1438,17 @@ uct_ib_md_open(const char *md_name, const uct_md_config_t *uct_md_config, uct_md
         goto err_dealloc_pd;
     }
 
-    status = uct_ib_md_parse_reg_methods(md, md_config);
+    status = uct_md_query(&md->super, &md_attr);
     if (status != UCS_OK) {
         goto err_destroy_umr_qp;
     }
 
-    status = uct_ib_md_parse_device_config(md, md_config);
+    status = uct_ib_md_parse_reg_methods(md, &md_attr, md_config);
```

Comment:
do we need to also change the order of destruction (in both error flow and good flow)

Reply:
din't change the order here.  just moved uct_md_quey() up before uct_ib_md_parse_reg_methods().